File: 7-kyu/computer-problem-series-#1--fill-the-hard-disk-drive/python/solution.py
def save(sizes, hd):
    if not sizes:
        return 0
    if sizes[0] > hd:
        return 0
    ret = []
    while sizes and sum(ret) < hd:
        ret.append(sizes.pop(0))
        if sizes and sizes[0] + sum(ret) > hd:
            break
    return len(ret)

File: 7-kyu/computer-problem-series-#1--fill-the-hard-disk-drive/javascript/solution.js
const save = (sizes, hd) => {
  if (!sizes) {
    return 0;
  }
  if (sizes[0] > hd) {
    return 0;
  }
  let ret = 0;
  let sum = 0;
  while (sizes.length > 0 && sizes[0] + sum <= hd) {
    sum += sizes.shift();
    ret++;
    if (sizes.length > 0 && sizes[0] + sum > hd) {
      break;
    }
  }
  return ret;
};
